### PR TITLE
[ONNX] Add einsum export

### PR DESCRIPTION
**Pull request by neginraoof**
Adding symbolic for onnx einsum as part of opset 12

### CODE REVIEW

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 5363f7eb:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=0dc38be40732c6d710d6710af243b851dc32a6c6&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base 0dc38be4 since Jan 28
  <details>
  <summary>
  Please rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=5363f7eb988925d35c40267d4512b8d587478e1d).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [pytorch_macos_10_13_py3_test](https://dr.pytorch.org/build-details.html?build_id=92098525) from Jan 28 until Jan 28



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Comment by yf225** [comment]
@houseroad Please let me know if I should ask someone else to review it. Thanks!

File: torch/onnx/symbolic_opset12.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+# EDITING THIS FILE? READ THIS FIRST!
+# see Note [Edit Symbolic Files] in symbolic_helper.py
+
+# This file exports ONNX ops for opset 12
+
+@parse_args('s', 'v')
+def einsum(g, equation, tensor_list):
+    return g.op("Einsum", tensor_list, equation_s=equation)
```
Comment (houseroad) [inline torch/onnx/symbolic_opset12.py]: How do we handle tensor_list here?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -2440,6 +2440,43 @@ def forward(self, input):
         self.run_test(DimModel(), empty_input)
         self.run_test(DimModel(), multi_dim_input)
 
+    @unittest.skip("Enable this once einsum supported in ORT")
+    @skipIfUnsupportedMinOpsetVersion(12)
+    def test_einsum(self):
+        class EinsumModel_1(torch.nn.Module):
```
Comment (houseroad) [inline test/onnx/test_pytorch_onnx_onnxruntime.py]: Nit: can we use meaningful suffix in test names, instead of _1, _2, and so on?

**Review by houseroad — APPROVED** [review]
Looks good, thanks.

**Review by houseroad — COMMENTED** [review]
Actually, tests are failing. Is it related?

**Comment by neginraoof** [comment]
> Actually, tests are failing. Is it related?
@houseroad Thanks.
Torchvision tests failed for ORT backend opset 12 tests. This is beacuse torchvision custom ops are registered as part of opset 11 only. I'll disabled those tests for opset 12. I'm going to send a fix for this in another PR.

File: torch/onnx/symbolic_opset12.py (hunk 1 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset12.py]: Thanks, I missed that. I fixed the symbolic and checked on onnx graph.

**Comment by tygrer** [comment]
I still cannot convert model from pytorch to onnx. Even though I has changed the source code of 
symbolic_opset12.py and symbolic_helper.py. The op of einsum is still not be recognized.
  File "/home/tanggy/anaconda3/envs/grab/lib/python3.6/site-packages/torch/onnx/__init__.py", line 148, in export
    strip_doc_string, dynamic_axes, keep_initializers_as_inputs)
  File "/home/tanggy/anaconda3/envs/grab/lib/python3.6/site-packages/torch/onnx/utils.py", line 66, in export
    dynamic_axes=dynamic_axes, keep_initializers_as_inputs=keep_initializers_as_inputs)
  File "/home/tanggy/anaconda3/envs/grab/lib/python3.6/site-packages/torch/onnx/utils.py", line 416, in _export
    fixed_batch_size=fixed_batch_size)
  File "/home/tanggy/anaconda3/envs/grab/lib/python3.6/site-packages/torch/onnx/utils.py", line 296, in _model_to_graph
    fixed_batch_size=fixed_batch_size, params_dict=params_dict)
  File "/home/tanggy/anaconda3/envs/grab/lib/python3.6/site-packages/torch/onnx/utils.py", line 135, in _optimize_graph
    graph = torch._C._jit_pass_onnx(graph, operator_export_type)
  File "/home/tanggy/anaconda3/envs/grab/lib/python3.6/site-packages/torch/onnx/__init__.py", line 179, in _run_symbolic_function
    return utils._run_symbolic_function(*args, **kwargs)
  File "/home/tanggy/anaconda3/envs/grab/lib/python3.6/site-packages/torch/onnx/utils.py", line 656, in _run_symbolic_function
    op_fn = sym_registry.get_registered_op(op_name, '', opset_version)
  File "/home/tanggy/anaconda3/envs/grab/lib/python3.6/site-packages/torch/onnx/symbolic_registry.py", line 91, in get_registered_op
    return _registry[(domain, version)][opname]
KeyError: 'einsum

**Pull request closed without merging**